src/shared/components/InputBase/InputBase.styles.ts

```diff
@@ -0,0 +1,103 @@
+import { colors, sizes, transitions, typography } from '@/shared/theme'
+import { css } from '@emotion/react'
+import styled from '@emotion/styled'
+
+type FormGroupProps = {
+  error?: boolean
+  disabled?: boolean
+}
+
+export const FormGroup = styled.div<FormGroupProps>`
+  max-width: 300px;
+  position: relative;
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'auto')};
+  input,
+  button {
+    transition: all ${transitions.timings.regular} ${transitions.easing};
+    background: none;
+    padding: ${sizes(3)};
+    font-size: ${typography.sizes.subtitle2};
+  }
+
+  input,
+  button,
+  label {
+    cursor: pointer;
+  }
+
+  ${({ disabled }) =>
+    disabled &&
+    css`
+      input,
+      button,
+      label {
+        pointer-events: none;
+      }
+    `};
+`
+
+export const LabelText = styled.span`
```

Comment:
There are visible rectangle corners on transition when label moves up. Maybe `background-color` transition could help. Or pseudoelements that will work as border-top and will shorten and make space when the label moves up.